assets/src/edit-story/components/header/buttons.js

```diff
@@ -55,19 +59,107 @@ function PreviewButton() {
+    let popup;
+    try {
+      popup = window.open('about:blank', PREVIEW_TARGET);
+      if (popup) {
+        popup.document.write('<!DOCTYPE html><html><head>');
+        popup.document.write('<title>');
+        popup.document.write(
+          escapeHTML(__('Generating the preview...', 'web-stories'))
+        );
+        popup.document.write('</title>');
+        popup.document.write('</head><body>');
+        // Output "waiting" message.
+        popup.document.write(
+          escapeHTML(
+            __('Please wait. Generating the preview...', 'web-stories')
+          )
+        );
+        // Force redirect to the preview URL after 5 seconds. The saving tab
+        // might get frozen by the browser.
+        popup.document.write(
+          `<script>
+            setTimeout(function() {
+              location.replace(${JSON.stringify(previewLink)});
+            }, 5000);
+          </script>`
+        );
+      }
+    } catch (e) {
+      // Ignore errors. Anything can happen with a popup. The errors
+      // will be resolved after the story is saved.
+    }
+
+    // @todo: See https://github.com/google/web-stories-wp/issues/1149. This
+    // has the effect of pushing the changes to a published story.
+    saveStory().then(() => {
+      let previewOpened = false;
+      if (popup && !popup.closed) {
+        try {
+          // The `popup.location.href` will fail if the expected window has
+          // been naviagted to a different origin.
+          if (popup.location.href) {
+            popup.location.replace(previewLink);
+            previewOpened = true;
+          }
+        } catch (e) {
+          // Ignore the errors. They will simply trigger the "try again"
+          // dialog.
+        }
+      }
+      if (!previewOpened) {
+        setPreviewLinkToOpenViaDialog(previewLink);
+      }
+    });
   };
+
+  const openPreviewLinkSync = (evt) => {
+    setPreviewLinkToOpenViaDialog(null);
+    // Ensure that this method is as safe as possible and pass the random
+    // target in case the normal target is not openable.
+    window.open(previewLinkToOpenViaDialog, PREVIEW_TARGET + Math.random());
+    evt.preventDefault();
+  };
+
   return (
-    <Outline onClick={openPreviewLink} isDisabled={isSaving}>
-      {__('Preview', 'web-stories')}
-    </Outline>
+    <>
+      <Outline onClick={openPreviewLink} isDisabled={isSaving}>
+        {__('Preview', 'web-stories')}
+      </Outline>
+      <Dialog
+        open={Boolean(previewLinkToOpenViaDialog)}
+        onClose={() => setPreviewLinkToOpenViaDialog(null)}
+        title={__('Open preview', 'web-stories')}
+        actions={
+          <>
+            <Primary onClick={openPreviewLinkSync}>
+              {__('Try again', 'web-stories')}
+            </Primary>
+            <Plain onClick={() => setPreviewLinkToOpenViaDialog(null)}>
+              {__('Cancel', 'web-stories')}
+            </Plain>
```

Comment:
Finally tested this flow.

These buttons might want a bit of spacing, and also the cancel text seem to be mapped to an uppercase version, which seems a bit shouty?

![Screen Shot 2020-04-13 at 10 21 22 AM](https://user-images.githubusercontent.com/637548/79127934-b41f9800-7d70-11ea-8011-9024da0a19eb.png)

Reply:
@miina, do you know where the current translations come from? And why would `'Cancel'` be mapped to `'CANCEL'`?

Reply:
As far as I can see then `Plain` is with a text-transform to `uppercase` within our buttons.

Reply:
Well, that explains it 🤦 

However, I don't think it looks good here. @samitron7?

Reply:
I'm going to merge this PR. The dialog styling issues should be resolved separately.